pkg/apis/planetscale/v2/vitessshard_types.go

```diff
@@ -315,6 +315,19 @@ type MysqldSpec struct {
 	ConfigOverrides string `json:"configOverrides,omitempty"`
 }
 
+// MysqlctldSpec configures the local mysqlctld gRPC server within a tablet.
+type MysqlctldSpec struct {
+	// Resources specify the compute resources to allocate for just the MySQL Control Daemon.
+	Resources corev1.ResourceRequirements `json:"resources"`
+
+	// ExtraFlags can optionally be used to override default flags set by the
+	// operator, or pass additional flags to mysqlctld. All entries must be
+	// key-value string pairs of the form "flag": "value". The flag name should
+	// not have any prefix (just "flag", not "-flag"). To set a boolean flag,
+	// set the string value to either "true" or "false".
+	ExtraFlags map[string]string `json:"extraFlags,omitempty"`
+}
```

Comment:
Unless I am missing something, this does not seem to be used anywhere. I think we should remove it to avoid unrequired changes

Reply:
I'll remove it if it indeed is not needed after I do some testing to confirm that things are now working.

Reply:
Coming back on what I said. I think it would be nice to keep that around so that people can use custom mysqlctld flags.